integration/experiment/ffnet/ffnet.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python3
+#
+#  Copyright (c) 2015 - 2024, Intel Corporation
+#  SPDX-License-Identifier: BSD-3-Clause
+#
+
+'''
+Helper functions for running CPU activity agent experiments.
+'''
+
+import argparse
+import os
+
+import geopmpy.agent
+
+from integration.experiment import launch_util
+from integration.experiment import common_args
+from integration.experiment import machine
+
+def setup_run_args(parser):
+    common_args.setup_run_args(parser)
+    parser.add_argument('--perf-energy-bias', dest='perf_energy_bias',
+                        action='store', default=0,
+                        help='Perf-Energy Bias [0-1] where 0 is perf-sensitive and 1 is energy-efficient')
+    parser.add_argument('--cpu-nn-path', dest='cpu_nn_path',
+                        action='store', default=None,
+                        help='Full path for the CPU NN')
+    parser.add_argument('--gpu-nn-path', dest='gpu_nn_path',
+                        action='store', default=None,
+                        help='Full path for the GPU NN')
+    parser.add_argument('--cpu-fmap-path', dest='cpu_freq_rec_path',
+                        action='store', default=None,
+                        help='Full path for the CPU Frequency Recommender Map')
+    parser.add_argument('--gpu-fmap-path', dest='gpu_freq_rec_path',
+                        action='store', default=None,
+                        help='Full path for the GPU Frequency Recommender Map')
+
+def report_signals():
+    return []
+
+def trace_signals():
+    return []
+
+def setup_env_paths(cpu_nn_path=None, cpu_fmap_path=None, gpu_nn_path=None, gpu_fmap_path=None):
+    if cpu_nn_path is None and gpu_nn_path is None:
+        raise RuntimeError('Must specify cpu-nn-path and/or gpu-nn-path when running ffnet experiment')
+
+    if cpu_nn_path is not None:
+        if os.path.exists(cpu_nn_path):
+            os.environ['GEOPM_CPU_NN_PATH'] = cpu_nn_path
+        else:
+            raise KeyError(f'File cpu-nn-path={cpu_nn_path} does not exist.')
```

Comment:
I would recommend raising a RuntimeError. A KeyError is not exactly what's happening here; a path does not exist (regardless of how it's being checked).

A similar comment applies below.